File: src/main/scala/jetbrains/zeppelin/api/rest/RestApiException.scala
package jetbrains.zeppelin.api.rest

class RestApiException(message: String = "", code: Int) extends Exception {
  override def getMessage: String = s"$message.\nError code: $code"
}

object RestApiException {
  def apply(message: String, code: Int): RestApiException = new RestApiException(message, code)
}

File: zeppelin/src/org/intellij/plugin/zeppelin/api/rest/RestApiException.kt
package org.intellij.plugin.zeppelin.api.rest

import org.intellij.plugin.zeppelin.models.ZeppelinException

data class RestApiException(val customMessage: String, val code: Int) : ZeppelinException() {
    override val message: String
        get() = "$customMessage.\nError code: $code"
}